velox/type/DecimalUtil.cpp

```diff
@@ -125,4 +125,14 @@ int32_t DecimalUtil::maxStringViewSize(int precision, int scale) {
   return rowSize;
 }
 
+int128_t DecimalUtil::big(int128_t value) {
```

Comment:
While this is the more generic implementation, its likely that this function is called for a vector of values which are all of type UUID, so all little endian. So doing the if condition each time is wasteful.

Do you want to make this templatized on kIsLittleEndian instead ? Another option is to make it a guarded function and add a debug assertion for folly::kIsLittleEndian.

Reply:
The compiler will optimize out the condition, since it is constant for a given platform. There is the same check done in `folly::Endian::big` btw